config/initializers/content_security_policy.rb

```diff
@@ -14,7 +14,12 @@
     # NOTE: This scirpt_src is overridden for all requests in ApplicationController
     # This is the baseline in case the override is ever skipped
     policy.script_src :self, "https://secure.gaug.es", "https://www.fastly-insights.com"
-    policy.style_src :self, "https://fonts.googleapis.com"
+    policy.style_src :self, "https://fonts.googleapis.com",
+      # This hardcoded value is the SHA256 of the style tag injected by turbo-rails ProgressBar.
+      # We don't want to rely on the nonce because it gets cached and then turbo can't reliably navigate.
+      # This is because new turbo responses and old cached responses will have different nonces, making CSP fail.
+      # You can regenerate this value by looking at script/turbo_rails_hash in this repo.
```

Comment:
Just to elaborate on this, when I've tested previously, the CSP nonce on Turbo loaded pages can clash with the nonce that exists on the current page, causing problems if any new nonced content comes with the new pages.

Reply:
I don't really like this part. I'll check how to properly handle this. Is it related to https://github.com/hotwired/turbo-rails/issues/458?

Reply:
Maybe, I was generally, maybe ideologically trying to avoid relying on nonces because of these ugly problems.

I think the "right way" is to remove the embedded stylesheet.

I prefer not using nonces because a cached nonce isn't a nonce.

Reply:
I checked github.com and they set style-src to 'unsafe-inline' which might allow this particular stylesheet, but they don't include nonces.

Reply:
pypi.org is another reference: No nonces, lots of sha256 in their CSP (none in github's)

Reply:
I'm afraid this will break some future updates. But on the other side, there is no need to block this on this issue. Once we will find out this causing troubles, we can check again and find better way to handle this.

Reply:
I think the most likely problem will be the turbo progress bar not showing because the CSS is broken. Maybe we could just add a copy of the css to our stylesheet? I'd just like to disable this injected stylesheet if we could.